packages/eas-cli/src/build/android/version.ts

```diff
@@ -165,3 +168,61 @@ async function writeBuildGradleAsync({
   const buildGradlePath = AndroidConfig.Paths.getAppBuildGradleFilePath(projectDir);
   await fs.writeFile(buildGradlePath, buildGradle);
 }
+
+/**
+ * Returns buildNumber that will be used for the next build. If current build profile
+ * has an 'autoIncrement' option set, it increments the version on server.
+ */
+export async function resolveRemoteVersionCodeAsync({
+  projectDir,
+  projectId,
+  exp,
+  applicationId,
+  buildProfile,
+}: {
+  projectDir: string;
+  projectId: string;
+  exp: ExpoConfig;
+  applicationId: string;
+  buildProfile: BuildProfile<Platform.ANDROID>;
+}): Promise<string> {
+  const remoteVersions = await AppVersionQuery.latestVersionAsync(
+    projectId,
+    AppPlatform.Android,
+    applicationId
+  );
+
+  const localVersions = await maybeResolveVersionsAsync(projectDir, exp, buildProfile);
+  let currentBuildVersion: string;
+  if (remoteVersions?.buildVersion) {
+    currentBuildVersion = remoteVersions.buildVersion;
+  } else {
+    if (localVersions.appBuildVersion) {
+      Log.warn(
+        'No remote versions are configured for this project, versionCode will be initialized based on the value from the local project.'
+      );
+      currentBuildVersion = localVersions.appBuildVersion;
+    } else {
+      Log.error(
+        `Remote versions are not configured and EAS CLI was not able to read the current version from your project. Use "eas build:version:set" to initialize remote versions.`
+      );
+      throw new Error('Remote versions are not configured.');
+    }
+  }
+  if (!buildProfile.autoIncrement) {
+    return currentBuildVersion;
+  }
+
+  const nextBuildVersion = getNextVersionCode(currentBuildVersion);
+  Log.log(`Incrementing versionCode ${currentBuildVersion} -> ${nextBuildVersion}.`);
```

Comment:
let's use ora here so we show a spinner when it's happening and a checkmark when it's done. text should read "Incrementing" when the request is in flight and "Incremented" when completed